src/client/context.rs

```diff
@@ -1880,3 +1896,20 @@ impl Context {
         rest::unpin_message(channel_id.into().0, message_id.into().0)
     }
 }
+
+impl Drop for Context {
+    /// Combines and sends all queued messages.
+    fn drop(&mut self) {
+        if !self.queue.is_empty() {
+            let _ = self.say(&self.queue);
+        }
+    }
+}
+
+/// Allows the `<<=` operator to be used to queue messages.
+#[cfg(feature="extras")]
+impl<'a> ShlAssign<&'static str> for &'a mut Context {
```

Comment:
This being static will only allow static strings to be passed in. This should be `ShlAssign<&'a str>`, so that users can pass slices of strings: `ctx <<= &string`